prow/plugins/lifecycle/lifecycle_test.go

```diff
@@ -17,133 +17,248 @@ limitations under the License.
+			body:                  "/lifecycle stale",
+			expectedNewLabels:     []string{lifecycleStaleLabel},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body: "/lifecycle frozen putrid stale",
+			name:                  "add lifecycle stale, have it",
+			body:                  "/lifecycle stale",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{lifecycleStaleLabel},
 		},
 		{
-			body: "/lifecycle frozen cancel",
+			name:                  "add lifecycle rotten, don't have it",
+			body:                  "/lifecycle rotten",
+			expectedNewLabels:     []string{lifecycleRottenLabel},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body:  "/lifecycle frozen",
-			added: []string{"lifecycle/frozen"},
+			name:                  "add lifecycle rotten, have it",
+			body:                  "/lifecycle rotten",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{lifecycleRottenLabel},
 		},
 		{
-			body:  "/lifecycle stale",
-			added: []string{"lifecycle/stale"},
+			name:                  "remove lifecycle frozen, have it",
+			body:                  "/remove-lifecycle frozen",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleFrozenLabel},
+			issueLabels:           []string{lifecycleFrozenLabel},
 		},
 		{
-			body:  "/lifecycle putrid",
-			added: []string{"lifecycle/putrid"},
+			name:                  "remove lifecycle frozen, don't have it",
+			body:                  "/remove-lifecycle frozen",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body:  "/lifecycle rotten",
-			added: []string{"lifecycle/rotten"},
+			name:                  "remove lifecycle stale, have it",
+			body:                  "/remove-lifecycle stale",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleStaleLabel},
+			issueLabels:           []string{lifecycleStaleLabel},
 		},
 		{
-			body:    "/remove-lifecycle frozen",
-			removed: []string{"lifecycle/frozen"},
+			name:                  "remove lifecycle stale, don't have it",
+			body:                  "/remove-lifecycle stale",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body:    "/remove-lifecycle stale",
-			removed: []string{"lifecycle/stale"},
+			name:                  "remove lifecycle rotten, have it",
+			body:                  "/remove-lifecycle rotten",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleRottenLabel},
+			issueLabels:           []string{lifecycleRottenLabel},
 		},
 		{
-			body:    "/remove-lifecycle stale\n/remove-lifecycle putrid\n/lifecycle frozen",
```

Comment:
Removed this test case because `GetIssueLabels` does not track removed labels in the fake github client:

https://github.com/kubernetes/test-infra/blob/f3787ed222bc128d06c8ed2f81f1464717133c97/prow/github/fakegithub/fakegithub.go#L193-L194

Without tracking the removed labels, three commands cannot be tested at once.

Reply:
A reason not to use the fake client?